ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,271 @@
+  of available [IntegrationTestScenario] resources provided by the user and takes a series of
+  actions to facilitate testing and promotion of that [Snapshot].
+- **DeploymentTarget Provisioning**: If any [IntegrationTestScenarios] specify a related
+  [Environment], then [integration-service] will clone that environment, in order to provision a new
+  bare environment for testing.
+- **App Deployment**: If any [IntegrationTestScenarios] specify a related
+  [Environment], then after cloning the environment, [integration-service] will initiate deployment
+  of a version of the application to be tested. See also [Decoupling Deployment].
+- **Snapshot Test Results Recording**: As PipelineRuns corresponding with the [IntegrationTestScenarios]
+  complete, [integration-service] updates test results on the [Snapshot].
+- **Source Repo Test Results Reporting**: As PipelineRuns corresponding with the [IntegrationTestScenarios]
+  complete, [integration-service] updates test results in the source repo service's API (like the
+  github "checks" API).
+- **Global Candidate List Maintenance**: When all PipelineRuns corresponding with the
+  [IntegrationTestScenarios] that are marked as non-optional complete and if the [Snapshot] is
+  designated as a post-merge [Snapshot], then [integration-service] updates the "global candidate
+  list" by way of the [Component] API.
+- **Release Initiation**: When all PipelineRuns corresponding with the [IntegrationTestScenarios]
+  that are marked as non-optional complete and if the [Snapshot] is designated as a post-merge
+  [Snapshot], and if there is a [ReleasePlan] in the namespace with auto-release enabled, then
+  [integration-service] updates the "global candidate list" by way of the [Component] API and
+  creates a [Release] CR to initiate a release in [release-service].
+
+Note that **Snapshot Construction** and **Global Candidate List Maintenance** are related: the only
+purpose of global candidate list maintenance is to support snapshot construction.
+
+## Decision
+
+- **App Deployment** and **DeploymentTarget Provisioning** will be dropped, per the [Decoupling Deployment] ADR.
+- **Release Initiation** will become the responsibility of [release-service]. See more below.
+- [integration-service] will continue to own **Test Initiation**, **Snapshot Test Results Recording**,
+  and **Source Repo Test Results Reporting**.
+- **Snapshot Construction** and **Global Candidate List Maintenance** will remain responsibilities
+  of [integration-service], but will undergo some changes to remove dependence on external APIs.
+
+### The Snapshot becomes the TestSubject
+
+[integration-service] will fork the [Snapshot] API to a new resource under its domain
+named [TestSubject].
+
+The purpose of this change is to make [integration-service] independent of other APIs. The name
+[TestSubject] is chosen to reflect what the [Snapshot] represented *in [integration-service]'s
+domain* in our original architecture.
```

Comment:
Moving away from a Snapshot as a centralized API will create challenges for visualizing the flow of artifacts through a system. If the integration service maintains its own representation, then there should be some way for us to visualize those details within its context. 

When we release, however, since we are breaking away from the common model, there would likely be _some other_ resource created by the release service. How can we enable continuity of resources throughout all controllers? As a user of the system, I would hope to be able to see information about a specific set of artifacts ... and follow its lifecycle through the system. For example, see what test cases might have returned as well as seeing that that specific set of artifacts is released successfully.

If we consider only the integration and release services, it seems manageable because we can just add annotations or something similar to illustrate the connections. If there are more controllers in the process, however, this pattern of adding annotations might become harder.